rotkehlchen/inquirer.py

```diff
@@ -920,6 +924,112 @@ def find_curve_pool_price(
+        underlying_token = None
+
+        farming_tokens = {token.farming_pool_token for token in read_gearbox_data_from_cache(token.chain_id)[0].values()}  # noqa: E501
+        with globaldb.conn.read_ctx() as cursor:
+            maybe_underlying_tokens = globaldb.fetch_underlying_tokens(
+                cursor=cursor,
+                parent_token_identifier=evm_address_to_identifier(
+                    address=token.evm_address,
+                    chain_id=token.chain_id,
+                    token_type=EvmTokenKind.ERC20,
+                ),
+            )
+
+        if token in farming_tokens:
+            farming_contract = EvmContract(
+                address=token.evm_address,
+                abi=node_inquirer.contracts.abi('GEARBOX_FARMING_POOL'),
+                deployed_block=0,  # not used here
+            )
+            try:
+                lp_token = farming_contract.call(node_inquirer=node_inquirer, method_name='stakingToken')  # noqa: E501
+                lp_token = deserialize_evm_address(lp_token)
+            except (RemoteError, BlockchainQueryError) as e:
+                log.error(f'Failed to query stakingToken method in Gearbox Pool. {e!s}')
+                return None
+        else:
+            lp_token = token.evm_address
+
+        lp_contract = EvmContract(
+            address=lp_token,
+            abi=node_inquirer.contracts.abi('GEARBOX_LP'),
+            deployed_block=0,  # not used here
+        )
+        if maybe_underlying_tokens is None or len(maybe_underlying_tokens) != 1:
+            # underlying token not recorded in the DB. Ask the chain
+            try:
+                remote_underlying_token = lp_contract.call(node_inquirer, 'underlyingToken')
+            except (RemoteError, BlockchainQueryError) as e:
+                log.error(f'Failed to query underlying token method in {node_inquirer.chain_name} Gearbox pool. {e!s}')  # noqa: E501
+                return None
+
+            try:
+                underlying_token_address = deserialize_evm_address(remote_underlying_token)
+            except DeserializationError:
+                log.error(f'underlying token call of {token.evm_address} returned invalid address {remote_underlying_token}')  # noqa: E501
+                return None
+
+            try:  # make sure it's in the global DB
+                underlying_token = get_or_create_evm_token(
+                    userdb=node_inquirer.database,
+                    evm_address=underlying_token_address,
+                    chain_id=token.chain_id,
+                    encounter=TokenEncounterInfo(
+                        description='Detecting Gearbox pool underlying tokens',
+                    ),
+                )
+            except NotERC20Conformant as e:
+                log.error(f'Error fetching {node_inquirer.chain_name} token {underlying_token_address} while detecting underlying tokens of {token.evm_address!s}: {e!s}')  # noqa: E501
```

Comment:
you are not returning here so the logic will continue and error later because you don't have the underlying token